From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: etil2jz <81570777+etil2jz@users.noreply.github.com>
Date: Fri, 22 Apr 2022 18:50:49 +0200
Subject: [PATCH] Fix hunger saturation depleting on peaceful

The food saturation value is depleted on peaceful, even though
the displayed hunger bar never goes down. Hunger (or any related value, including saturation)
should not go down on peaceful. See https://bugs.mojang.com/browse/MC-31819.

diff --git a/src/main/java/net/minecraft/world/entity/player/Player.java b/src/main/java/net/minecraft/world/entity/player/Player.java
index 61597ebe2f9faff43994c475074b87d11905e582..6f6fdab36ed5a912ad655a466dd16644eb7139fc 100644
--- a/src/main/java/net/minecraft/world/entity/player/Player.java
+++ b/src/main/java/net/minecraft/world/entity/player/Player.java
@@ -1901,6 +1901,7 @@ public abstract class Player extends LivingEntity {
     }
 
     public void causeFoodExhaustion(float f, EntityExhaustionEvent.ExhaustionReason reason) {
+        if (this.level.getDifficulty() == Difficulty.PEACEFUL) return; // DivineMC - fix hunger saturation depleting on peaceful
         // CraftBukkit end
         if (!this.abilities.invulnerable) {
             if (!this.level.isClientSide) {
